packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -141,7 +141,7 @@ export function defineCustomElement(
   class VueCustomElement extends VueElement {
     static def = Comp
     constructor(initialProps?: Record<string, any>) {
-      super(Comp, initialProps, hydrate)
+      super(Comp, initialProps, hydrate, options.shadowRoot)
```

Comment:
If this `options.shadowRoot` is passed directly to `sfc` and uses `script setup`, how will it be set?

Reply:
You can use this reference configuration method, use a config as a parameter to pass in the control #4404 , and if possible, I would like to add this pr change idea to [unplugin-vue-ce](https: //github.com/baiwusanyu-c/unplugin-vue-ce)

Reply:
Thank you very much for your interest in this PR. I will seriously consider your suggestion, and I will think further about this PR. Thank!